src/testcases/org/apache/poi/ss/formula/TestWorkbookEvaluator.java

```diff
@@ -606,4 +608,38 @@ private void testIFEqualsFormulaEvaluation_evaluateAllFormulaCells(
         
         testIFEqualsFormulaEvaluation_teardown(wb);
     }
+
+    @Test
+    public void testSheetNameReference() throws IOException {
+        Workbook wb = new HSSFWorkbook();
+
+        Sheet s1 = wb.createSheet("Sheet1");
+        Cell cell1 = CellUtil.getCell(CellUtil.getRow(0, s1), 0);
+        cell1.setCellValue(123);
+        Sheet s2 = wb.createSheet();
+        Cell cell2 = CellUtil.getCell(CellUtil.getRow(0, s2), 0);
+
+        FormulaEvaluator eval = wb.getCreationHelper().createFormulaEvaluator();
+        {
+            cell2.setCellFormula("Sheet1!A1");
+            CellValue value = eval.evaluate(cell2);
+            assertEquals(123d, value.getNumberValue(), 0);
+        }
+
+        for (int i = 256; i <= 65535; i++) {
+            char c = (char) i;
+            String name = String.valueOf(c);
+            try {
+                WorkbookUtil.validateSheetName(name);
+            } catch (IllegalArgumentException e) {
+                continue;
+            }
+            wb.setSheetName(0, name);
+            cell2.setCellFormula(name + "!A1");
+            CellValue value = eval.evaluate(cell2);
+            assertEquals(123d, value.getNumberValue(), 0);
```

Comment:
Any assertions inside a for-loop should hint at what loop iteration resulted in the failure.

    assertEquals(String.format("With sheet name: '%s' (%i)", name, i), 123d, value.getNumberValue(), 0);